frontend/packages/kubevirt-plugin/src/selectors/pod/selectors.ts

```diff
@@ -42,15 +42,22 @@ export const findPodWithOneOfStatuses = (pods: PodKind[], statuses: string[]) =>
     return statuses.some((status) => status === phase);
   });
 
-export const findVMPod = (pods: PodKind[], vm: VMKind, podNamePrefix: string) => {
+export const findVMPod = (pods: PodKind[], vm: VMKind) => {
   if (!pods) {
     return null;
   }
-  const prefix = `${podNamePrefix}${getName(vm)}-`;
-  const prefixedPods = pods.filter(
-    (p) => getNamespace(p) === getNamespace(vm) && getName(p).startsWith(prefix),
+
+  let prefixedPods = pods.filter(
+    (p) => getNamespace(p) === getNamespace(vm) && getOwnerReferences(p)[0].name === getName(vm),
```

Comment:
Can we rather find the exact reference? To be safer we probably should use the prefix as well (please see `findConversionPod` for code example). There might be actually also be conversion pod with the same VM reference. Even though these two pods should never exist at the same time, we should also check the prefix to be safe.